### PR TITLE
Feat: Add button component

### PR DESCRIPTION
**Pull request by starver20**

### CODE REVIEW

File: components/button/button.html [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+              The Button component is used to trigger an action or event, such
+              as submitting a form, opening a dialog, canceling an action, or
+              performing a delete operation.
+            </p>
+            <div class="component-type">Button Types</div>
+
+            <div class="component-example">
+              <div class="button-container">
+                <button class="btn primary">Primary</button>
+                <button class="btn secondary">Secondary</button>
```
Comment (Husain01) [inline components/button/button.html:79]: I think secondary doesn't have the hover effect being applied due to same color

Comment (gadgetfather) [reply components/button/button.html:79]: Yea i was going to say same

File: components/button/button.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+.warning {
+  background-color: var(--primary-warning-color);
+  color: black;
+}
+
+.danger {
+  background-color: var(--primary-danger-color);
+}
+
+.primary:hover {
```
Comment (gadgetfather) [inline components/button/button.css:33]: You can try adding a transition of 0.2s to make it look smoother.

**Review by gadgetfather — APPROVED** [review]
well written class! good to merge

**Review by Husain01 — APPROVED** [review]
Looks good to go

**Review by mohammed-sas — APPROVED** [review]
your font-size seems to be smaller when i opened in code sandbox. increase the size . other than that you are good to merge

File: components/button/button.html (hunk 1 above)
Comment (starver20) [reply components/button/button.html:79]: I have intentionally not given any hover to secondary.

**Pull request merged**